src/main/java/org/gridsuite/ds/server/service/DynamicSimulationWorkerService.java

```diff
@@ -159,6 +186,23 @@ public void preRun(DynamicSimulationRunContext runContext) {
         runContext.setDynamicModelContent(dynamicModel);
         runContext.setEventModelContent(eventModel);
         runContext.setCurveContent(curveModel);
+
+        // create a working folder for this run
+        Path localDir = getComputationManager().getLocalDir();
+        Path workDir;
+        try {
+            workDir = Files.createTempDirectory(localDir, "dynamic_simulation_");
+            runContext.setWorkDir(workDir);
+
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+
+        // enrich dump parameters
+        Path dumpDir = workDir.resolve("dump");
+        FileUtil.createDirectory(dumpDir);
+        DynawoSimulationParameters dynawoSimulationParameters = parameters.getExtension(DynawoSimulationParameters.class);
+        dynawoSimulationParameters.setDumpFileParameters(DumpFileParameters.createExportDumpFileParameters(dumpDir));
```

Comment:
The extension is created in the previous step https://github.com/gridsuite/dynamic-simulation-server/blob/main/src/main/java/org/gridsuite/ds/server/service/parameters/impl/ParametersServiceImpl.java#L109, so it is never null.

Otherwise, throw another exception is not better than NPE, anyway the program must be stopped, the difference is instead of returning to end user a technical exception, we return a functional exception.